packages/graphql-fixtures/src/fill.ts

```diff
@@ -198,8 +201,21 @@ function unwrapThunk<T>(value: Thunk<T>, details: ResolveDetails): T {
   return isResolver(value) ? value({...details, type: unwrappedType}) : value;
 }
 
-function withRandom<T>(keypath: FieldDetails[], func: () => T) {
-  faker.seed(seedFromKeypath(keypath.map(({responseName}) => responseName)));
+function keyPathElement(responseName: string, fieldIndex: number | undefined) {
+  return fieldIndex == null ? responseName : `${responseName}[${fieldIndex}]`;
+}
+
+// we need to set a seedOffset when filling fields that are indexed leafs in the
+// graph, for indexed objects in the graph their key path will use the parent
+// field index instead.
+function withRandom<T>(keypath: FieldDetails[], func: () => T, seedOffset = 0) {
```

Comment:
I am confused, why do you need both this offset and the index? I guess I don't understand what the comment above is trying to tell me.

Reply:
`seedOffset` is for when the field we're filling is itself indexed (i.e., `details` from `createValue`) because otherwise it just relies on the parent key path for seeding which would be the same for all of its peer nodes.